src/Types/TypedValidator.purs

```diff
@@ -46,38 +46,49 @@ import Types.Scripts
 -- it suffices.
 -- | A typeclass that associates a type standing for a connection type with two
 -- | types, the type of the redeemer and the data script for that connection type.
+class ValidatorTypes :: Type -> Type -> Type -> Constraint
 class
-  ( DatumType a b
-  , RedeemerType a b
+  ( DatumType validator datum
+  , RedeemerType validator redeemer
   ) <=
-  ValidatorTypes (a :: Type) (b :: Type)
-  | a -> b
+  ValidatorTypes (validator :: Type) (datum :: Type) (redeemer :: Type)
```

Comment:
The old kind signatures are redundant now since we have the standalone kind signature